BFWControls/Sample/Transition/TransitionViewController.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  TransitionViewController.swift
+//  BFWControls
+//
+//  Created by Andy Kim on 12/5/17.
+//  Copyright © 2017 BareFeetWare. All rights reserved.
+//
+
+import UIKit
+
+
+class TransitionViewController: UIViewController {
+
+	let interactiveTransition = TranslationAnimationController()
+	
+	enum SegueIdentifier: String {
+		case interactiveSegue
+	}
+	
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+
+	
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+		guard let segueIdentifier = SegueIdentifier(rawValue: segue.identifier!)
+			else { return }
+		switch segueIdentifier {
+		case .interactiveSegue:
+			let viewController = segue.destination
+			viewController.transitioningDelegate = interactiveTransition
+		}
+    }
+
+
+	@IBAction func handlePanGesture(_ pan: UIPanGestureRecognizer) {
+		let translation = pan.translation(in: pan.view!)
+		let d =  translation.y / pan.view!.bounds.height
+		switch pan.state {
+		case .began:
+			interactiveTransition.isInteractive = true
+			interactiveTransition.direction = .down
+			performSegue(withIdentifier: SegueIdentifier.interactiveSegue.rawValue, sender: self)
```

Comment:
If you conform this view controller to `SegueIdentifierType`, `performSegue` can use the enum `.interactiveSegue` directly instead of `SegueIdentifier.interactiveSegue.rawValue`